In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statistics as st

Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

<i>This paragraph is misleading and ambiguous.</i>

In [2]:
names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Oliver', 'Cindy' ]

In [3]:
df = pd.DataFrame(index = names)

In [4]:
df['age'] = [14, 12, 11, 10, 8,8,6]

In [5]:
df

age
Greg     14
Marcia   12
Peter    11
Jan      10
Bobby     8
Oliver    8
Cindy     6

In [6]:
np.median(df['age'])

10.0

In [7]:
np.mean(df['age'])

9.857142857142858

In [8]:
st.mode(df['age'])

8

In [9]:
np.var(df['age'], ddof=1)

7.476190476190475

In [10]:
np.std(df['age'])

2.531435020952764

In [11]:
np.std(df['age'])/np.sqrt(len(df['age']))    #se = s / (n ** 0.5)

0.9567925036515135

Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?<br />
<i>I would consider using the value of mode because it is a repetitive set of values that can impact the rest of the sample. <br />In the variance, I will pick the wides range of the variation because I need to have an idea of how wide is the distribution.</i>

Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?<br />
<i>The mean increased values, the variance, the standard deviation, and error decreased as well.</i>

In [12]:
df.loc['Cindy','age'] += 1

In [13]:
df

age
Greg     14
Marcia   12
Peter    11
Jan      10
Bobby     8
Oliver    8
Cindy     7

In [14]:
np.median(df['age'])

10.0

In [15]:
np.mean(df['age'])

10.0

In [16]:
st.mode(df['age'])

8

In [17]:
np.var(df['age'], ddof=1)

6.333333333333333

In [18]:
df.std()

age    2.516611
dtype: float64

In [19]:
df.std()/np.sqrt(len(df['age']))

age    0.95119
dtype: float64

Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [20]:
df2 = df.rename(index={'Oliver': 'Jessica'})

In [21]:
df2.loc['Jessica', 'age'] -= 7

In [22]:
df2

age
Greg      14
Marcia    12
Peter     11
Jan       10
Bobby      8
Jessica    1
Cindy      7

In [23]:
np.median(df2['age'])

10.0

In [24]:
np.mean(df2['age'])

9.0

In [25]:
# formula st.mode(df2['age']) give error
df2.mode()

age
0    1
1    7
2    8
3   10
4   11
5   12
6   14

In [26]:
np.var(df2['age'], ddof=1)

18.0

In [27]:
np.std(df2['age'])

3.927922024247863

In [28]:
st.stdev(df2['age'])/np.sqrt(len(df2['age']))

1.6035674514745462

<i>With the replacement of Jessica for Oliver, there was no mode, the mean decrease but the variance, the standard deviation, and the standard error increased.<br />Because the mode disappears, I will pick mean as a central tendency.</i>

On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [29]:
magazines = ['TV Guide', 'Entertainment Weekly', 'Pop Culture Today']

In [30]:
anniversary = pd.DataFrame(index=magazines)

In [31]:
anniversary['popularity'] = [20, 23, 17]

In [32]:
anniversary

popularity
TV Guide                      20
Entertainment Weekly          23
Pop Culture Today             17

In [35]:
np.mean(anniversary['popularity'])

20.0

In [37]:
np.std(anniversary['popularity'])/np.sqrt(len(anniversary['popularity']))

1.414213562373095

The percentage of adult Americans that were Brady Bunch fans on the 50th anniversary is 20%  with a 1.41% of error